### PR TITLE
Update acoustics

### PR DESCRIPTION
**Pull request by oyssolbo**
Added reading the DMA, however is a bit unsure how well it would work. Bit unsure as the documentation and implementation vary from MCU to MCU, and haven't found anything for STM32F7xxxx. Potential future bug(!!) but could be solved in a future issue/PR

### CODE REVIEW

File: Source/main.cpp [hunk 1]
```diff
@@ -434,66 +461,98 @@ static void MX_GPIO_Init(void)
...
+ */
+static void read_ADC(float32_t* p_data_hyd_port, float32_t* p_data_hyd_starboard,
+            float32_t* p_data_hyd_stern){
+  /**
+   * Reading the data. Dropping the last couple of datapoints, since
+   * 4096 % 3 = 1. Reducing the number of datapoints reduces the 
+   * accuracy of the analysis, however prevents out-of-range error
+   */
+  for(int i = 0; i < DSP_CONSTANTS::DMA_BUFFER_LENGHT - 
+        NUM_HYDROPHONES, i += NUM_HYDROPHONES){
```
Comment (chrstrom) [inline Source/main.cpp:501]: Should be a ; after NUM_HYDROPHONES

File: Source/main.cpp [hunk 2]
```diff
@@ -434,66 +461,98 @@ static void MX_GPIO_Init(void)
...
 }
 
 
+/**
+ * @brief Function to log the errors
+ * Could also be interesting to implement a timestamp to detect when the
+ * error occurs. This could be done in the future, but is not a priority
+ * as of 14.12.2020
+ * 
+ * NOTE: Could be improved by using a txt-file or other log, or send the number
```
Comment (chrstrom) [inline Source/main.cpp:526]: I agree, logging the errors on the Xavier seems to be a good option. Having the time synced with the Xavier would also allow us to pinpoint where eventual errors appear, in relation to everything else that is going on in our system.

Comment (chrstrom) [reply Source/main.cpp:526]: But yeah, as you write, this is still early phase, so I think figuring out the communication protocol between the Xavier and the board is the main priority in that regard.

File: Source/main.cpp (hunk 1 above)
Comment (chrstrom) [inline Source/main.cpp:501]: Should be a ; instead of a , after NUM_HYDROPHONES

**Review by chrstrom — CHANGES_REQUESTED** [review]
I can't say I know how everything should be implemented on the STM board, but the code is well documented and clean enough to be passed on for now, given the minor fix that I commented on 😉

File: Source/main.cpp (hunk 1 above)
Comment (oyssolbo) [reply Source/main.cpp]: Corrected. Thanks for noticing such a small typo

**Pull request merged**